src/DataSetUtils.pas

```diff
@@ -22,8 +22,17 @@ class function TDataSetUtils.CreateDataSetField(DataSet: TDataSet;const FieldNam
 
 class function TDataSetUtils.CreateField(DataSet: TDataSet;
   FieldType: TFieldType; const FieldName: string; ASize: Integer; ADisplayWidth: Integer): TField;
+var
+  fieldClass: TFieldClass;
 begin
-    Result:= DefaultFieldClasses[FieldType].Create(DataSet);
+    fieldClass := DefaultFieldClasses[FieldType];
+    if (fieldClass = nil) then
+    begin
+      fieldClass := TStringField;
+      ASize := 255;
+    end;
+
+    Result:= fieldClass.Create(DataSet);    Result:= DefaultFieldClasses[FieldType].Create(DataSet);
```

Comment:
this looks a little suspicious.

Reply:
It doesn't the same, because DefaultFieldClasses for ftUnknown and ftCursor returns nil, at least on delphi 7. In my case the server returns null on a json Field and the result is ftUnknowm generating an access violation

Reply:
So this doesn't work for you?

This is what is being replaced:
```pascal
Result := DefaultFieldClasses[FieldType].Create(DataSet);
```
To me this looks like the following should work.  At least, I don't see why it wouldn't.
```pascal
fieldClass := DefaultFieldClasses[FieldType];
if (fieldClass = nil) then
begin
  fieldClass := TStringField;
  ASize := 255;
end;

// Result := fieldClass.Create(DataSet);
Result := DefaultFieldClasses[FieldType].Create(DataSet);
```

It appears to me that you do not need `Result := fieldClass.CreateDataSet` since `Result` is then immediately replaced by `Result := DefaultFieldClasses[FieldType].Create(DataSet);`.  Or, do you really want the first `Result` since it will never be `nil`?  I am not seeing how the new logic prevents function `CreateFields` from possibly returning `nil`.  I am not understanding how the added logic prevents
```pascal
Result := DefaultFieldClasses[FieldType].Create(DataSet);
```
from returning `nil`;

I am not understanding how this change is preventing the issue you are experiencing and am hoping I am going to learn something new here.

Reply:
** This is my personal account, I'll use this account from now on **

This is DefaultFieldClasses declaration on delphi
```delphi
   DefaultFieldClasses: array[TFieldType] of TFieldClass = (
    nil,                { ftUnknown }
    TStringField,       { ftString }
    TSmallintField,     { ftSmallint }
    TIntegerField,      { ftInteger }
    // ... other field types
);
```
So, when the function *CreateField* is called passing the parameter *FieldType* as *ftUnknown*
the statement 
```delphi 
DefaultFieldClasses[FieldType]
```
returns a nil value.

if you do this
```delphi
Result := DefaultFieldClasses[FieldType].Create(DataSet);
```
The application throws an Access Violation exception because nil doesn't have the *constructor* Create

I did this: I've created a local variable *fieldClass: TFieldClass;* and this local variable receives the field class, in my case nil, then I verify if this local variable is nil, if it is nil, I change the value of the local variable *fieldClass* to a TStringField.
```delphi
 if (fieldClass = nil) then
 begin
     fieldClass := TStringField;
     ASize := 255;
end;
```
So the result will receive my local variable constructor create which will never be a nil value, avoiding access violation:

```delphi
Result := fieldClass.Create(DataSet);
```
I believe that a String field won't change the use in anyone's code, the other option would be do not create any field, but I think this way is going to be more annoying test if the field exists or not every time you try to access the field.

Reply:
Now I understood the problem:
this line:
```delphi
Result:= fieldClass.Create(DataSet);    Result:= DefaultFieldClasses[FieldType].Create(DataSet);
```
Should be just
```delphi
Result:= fieldClass.Create(DataSet);    
```
Something went wrong here... Sorry I'll correct this...

Reply:
Now I understood the problem:
this line:
```delphi
Result:= fieldClass.Create(DataSet);    Result:= DefaultFieldClasses[FieldType].Create(DataSet);
```
Should be just
```delphi
Result:= fieldClass.Create(DataSet);    
```
Something went wrong here... Sorry I'll correct this...

Reply:
From my understanding of how things work, the last line was defeating the purpose of the code change.  The first half of the line made sense.  The later half did not.